src/rustfmt_diff.rs

```diff
@@ -97,15 +97,28 @@ pub fn make_diff(expected: &str, actual: &str, context_size: usize) -> Vec<Misma
     results
 }
 
+// A representation of how to write output.
+pub enum PrintType {
+    Fancy, // want to output color and the terminal supports it
+    Basic, // do not want to output color or the terminal does not support color
```

Comment:
@nrc: Here's the first set of comments you requested.

After this PR, I'm going to spin up a new one to merge `print_diff_fancy` and `print_diff_basic` by adding more implementation to the enum. I think the enum is too abstract like this.